### PR TITLE
#170529662 Design shareable sidebar

### PR DESCRIPTION
**Pull request by NiyoEric**
#### What does this PR do?
The PR helps to share common sidebar. 

#### Description of Task to be completed?
**Expected**
- Expect to have a common top sidebar 
- Expect to divided right and left sidebar

#### How should this be manually tested?
- `git clone https://github.com/andela/team-odd-bn-frontend.git`
- `cd team-odd-bn-frontend`
- `git checkout ch-design-sidebar-170529662`
- `npm install` packages
- `npm run dev` 
- Visit the dashboard page. For example  `URL/dashboard`
- Test responsiveness

#### Any background context you want to provide?
I have designed the re-usable sidebar. Top, right and left sides are an independent component. You are expected to add content in the right side.

#### What are the relevant pivotal tracker stories?
[#170529662](https://www.pivotaltracker.com/story/show/170529662)

#### Screenshots (if appropriate)

Desktop view
Screen Shot 2020-01-06 at 07 22 11

Ipad view 
Screen Shot 2020-01-06 at 07 23 15

Ipad view side menu
Screen Shot 2020-01-06 at 07 23 23

Mobile view
Screen Shot 2020-01-06 at 07 23 34

Mobile view side menu
Screen Shot 2020-01-06 at 07 23 45

### CODE REVIEW

File: src/assets/css/sidenav/DrawerToggleButton.scss [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+.toggle-button {
+  outline: none;
+}
+
+.toggle-button__line {
+  width: 30px;
+  height: 3px;
+  background-color: black;
+}
```
Comment (hezronkimutai) [inline src/assets/css/sidenav/DrawerToggleButton.scss]: eol

File: src/assets/css/sidenav/sidebar.scss [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+  cursor: pointer;
+}
+
+/*******************************
+  Left and right sidebar style
+*******************************/
+
+.top-side {
+  @extend %top-side-header;
+  // @extend %z-index-value;
```
Comment (hezronkimutai) [inline src/assets/css/sidenav/sidebar.scss]: Remove the comment here

File: src/assets/css/sidenav/tooltip.scss [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  width: 120px;
+  top: 150%;
+  left: 50%;
+  margin-left: -60px;
+}
+
+
+.tooltip:hover .tooltiptext {
+  visibility: visible;
+}
```
Comment (hezronkimutai) [inline src/assets/css/sidenav/tooltip.scss]: eol

File: src/config/index.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export const BASE_URL = 'https://team-odd-bn-backend-staging.herokuapp.com/api/v1';
+export const ICON = 'https://res.cloudinary.com/hezzie/image/upload/v1577645239/Screen_Shot_2019-12-29_at_20.46.06_vklvux.png';
+export const FACEBOOK_CALLBACK_URL = 'https://team-odd-bn-backend-staging.herokuapp.com/api/v1/auth/facebook';
+export const GOOGLE_CALLBACK_URL = 'https://team-odd-bn-backend-staging.herokuapp.com/api/v1/auth/google';
```
Comment (hezronkimutai) [inline src/config/index.js]: Remember to move this data to `.env`

**Review by hezronkimutai — CHANGES_REQUESTED** [review]
Nice initiative @NiyoEric , Your work will make our subsequent tasks easier, I left some comments, please view them.

**Review by nziokaivy — APPROVED** [review]
Great work @NiyoEric !!

**Review by hezronkimutai — APPROVED** [review]

**Review by hezronkimutai — COMMENTED** [review]
Hi, @NiyoEric , Can you reduce the height of the right side to 90%  since footer already takes 100%. Also remove the z-index  styling on the right side as it disables buttons and input fields.

File: src/App.js [hunk 5]
```diff
@@ -19,8 +25,7 @@ const App = () => (
         <Switch>
           <Route path="/signin" exact component={Signin} />
           <Route path="/social/auth/success" exact component={SocialLogin} />
-          <Route path="/dashboard" exact component={Dashboard} />
-          <Route path="/dashboard" exact component={VerifyEmail} />
```
Comment (shemaeric) [inline src/App.js]: @NiyoEric why did you remove this Route.

**Review by shemaeric — CHANGES_REQUESTED** [review]
@NiyoEric I can't be able to test your implementation as described on your `PR description`.
The route for the `dashboard` doesn't exist.

Please checkout on this.

File: src/App.js (hunk 5 above)
Comment (NiyongaboEric) [reply src/App.js]: Let me add it again.

Comment (NiyongaboEric) [reply src/App.js]: Try to start a local server again and visit `http://localhost:4000/dashboard`.

**Review by shemaeric — APPROVED** [review]
